scripts/kind-e2e/lib_operator_deploy_subm.sh

```diff
@@ -5,40 +5,90 @@ if [ "${0##*/}" = "lib_operator_deploy_subm.sh" ]; then
     exit 1
 fi
 
+function create_subm_vars() {
+  # FIXME A better name might be submariner-engine, but just kinda-matching submariner-<random hash> name used by Helm/upstream tests
+  deployment_name=submariner
+  operator_deployment_name=submariner-operator
+  engine_deployment_name=submariner-engine
+  routeagent_deployment_name=submariner-routeagent
+  broker_deployment_name=submariner-k8s-broker
+
+  clusterCIDR_cluster1=10.244.0.0/16
+  clusterCIDR_cluster2=10.245.0.0/16
+  clusterCIDR_cluster3=10.246.0.0/16
+  serviceCIDR_cluster2=100.95.0.0/16
+  serviceCIDR_cluster3=100.96.0.0/16
+  serviceCIDR_cluster1=100.94.0.0/16
+  natEnabled=false
+
+  subm_engine_image_repo=local
+  subm_engine_image_tag=local
+
+  # FIXME: Actually act on this size request in controller
+  subm_engine_size=3
+  subm_colorcodes=blue
+  subm_debug=false
+  subm_broker=k8s
+  ce_ipsec_debug=false
+  ce_ipsec_ikeport=500
+  ce_ipsec_nattport=4500
+
+  subm_ns=submariner-operator
+  subm_broker_ns=submariner-k8s-broker
+}
+
+function preinstall_cleanup_subm {
+    context=$1
+    delete_subm_pods $context submariner-operator
+}
+
 function get_subctl() {
     test -x /go/bin/subctl && return
     curl -L https://github.com/submariner-io/submariner-operator/releases/download/v0.0.2/subctl-v0.0.2-linux-amd64 \
          -o /go/bin/subctl
     chmod a+x /go/bin/subctl
 }
 
-function add_subm_gateway_label() {
-  kubectl label node $context-worker "submariner.io/gateway=true" --overwrite
+function deploytool_prereqs() {
+    get_subctl
 }
 
-function del_subm_gateway_label() {
-  kubectl label node $context-worker "submariner.io/gateway-" --overwrite
+function setup_broker() {
+    context=$1
+    if kubectl --context=$context get crd clusters.submariner.io > /dev/null 2>&1; then
+        echo Submariner CRDs already exist, skipping broker creation...
+    else
```

Comment:
I believe the if is not necessary anymore and subctl can be called twice to setup the broker without crash.

Reply:
Okay, changed in the latest version - thanks!